opendevin/action/fileop.py

```diff
@@ -39,11 +46,16 @@ class FileWriteAction(ExecutableAction):
     action: str = ActionType.WRITE
 
     def run(self, controller) -> FileWriteObservation:
-        whole_path = resolve_path(controller.workdir, self.path)
-        with open(whole_path, "w", encoding="utf-8") as file:
-            file.write(self.content)
-        return FileWriteObservation(content="", path=self.path)
+        if isinstance(controller.command_manager.sandbox, E2BBox):
```

Comment:
This feels leaky to me. But passing `controller` into `run` is already kind of leaky. Curious if you have any ideas on how to improve this.

Maybe `sandbox` should handle all the `run` logic...

Reply:
Yeah I agree this doesn't feel right. Happy to refactorize this